generator/internal/genclient/model.go

```diff
@@ -74,6 +74,8 @@ type API struct {
 type APIState struct {
 	// ServiceByID returns a service that is associated with the API.
 	ServiceByID map[string]*Service
+	// MethodByID returns a method that is associated with the API.
```

Comment:
Unrelated, but since we are in that type: do we need `APIState`?  Seems like it is always tied to one `API` instance, and it is (like `API`) immutable once the parser finishes with it.

Should we just merge the two structs? (in a future PR)

Reply:
In my mind the main reason this is seperate today is because it is what is needed to be passed into a lot of the codec functions. For that reason having a smaller type for lookups in nice vs passing in the full API everytime. Thoughts?

Reply:
You mean it has a narrower interface?  You are right, that is a Good Thing :tm:   But that argues for passing an interface to the codec methods, instead of an specific implementation.

We should not do design via review comments.  I think we have enough for a future discussion.